src/firmware/shared/math/polynomial_2d_test.cpp

```diff
@@ -2,460 +2,489 @@ extern "C"
+        float arc_length_values[17] = {0,      168.5,  314.3,  437.58, 538.49, 617.19,
+                                       673.86, 708.65, 721.73, 730.46, 760.33, 811.43,
+                                       883.60, 976.68, 1090.5, 1224.9, 1379.8};
+        ArcLengthParametrization_t arc_length_parametrization = {
+            .t_values          = t_values,
+            .arc_length_values = arc_length_values,
+            .num_values        = 17};
+
+        // Check some arc length points that require no interpolation
+        expectVector2dEqual({.x = 119.9, .y = 717.00},
+                            shared_polynomial2d_getPositionAtArcLengthOrder3(
+                                poly, 0, arc_length_parametrization),
+                            0.01);
+        expectVector2dEqual({.x = 128.3, .y = 652.2},
+                            shared_polynomial2d_getPositionAtArcLengthOrder3(
+                                poly, 1379.8, arc_length_parametrization),
+                            0.01);
+        expectVector2dEqual({.x = 70.1, .y = -3.0},
+                            shared_polynomial2d_getPositionAtArcLengthOrder3(
+                                poly, 721.73, arc_length_parametrization),
+                            0.01);
+
+        // Check some arc length points that should be determined via interpolation
+        expectVector2dEqual({.x = 111.81, .y = 588.75},
+                            shared_polynomial2d_getPositionAtArcLengthOrder3(
+                                poly, 126.38, arc_length_parametrization),
+                            0.01);
+        expectVector2dEqual({.x = 97.479, .y = 370.5},
+                            shared_polynomial2d_getPositionAtArcLengthOrder3(
+                                poly, 345.12, arc_length_parametrization),
+                            0.01);
+    }
 
-TEST_F(Polynomial2dTest, get_t_value_on_arc_length_below_arc_lengths_in_parametrization)
-{
-    Polynomial2dOrder1_t poly = {
-        .x = {.coefficients = {1, 0.1}},
-        .y = {.coefficients = {1, 0.1}},
-    };
+    TEST_F(Polynomial2dTest, get_t_value_at_arc_length_on_straight_line_1_values)
+    {
+        // This function should probably never be used like this, but this checks that
+        // we're at least executing defined behavior
+
+        Polynomial2dOrder1_t poly = {
+            .x = {.coefficients = {1, 0.1}},
+            .y = {.coefficients = {1, 0.2}},
+        };
+
+        float t_values[1]                                     = {0.5};
+        float arc_length_values[1]                            = {99};
+        ArcLengthParametrization_t arc_length_parametrization = {
+            .t_values          = t_values,
+            .arc_length_values = arc_length_values,
+            .num_values        = 1};
+
+        EXPECT_FLOAT_EQ(0.5, shared_polynomial2d_getTValueAtArcLengthOrder1(
+                                 poly, 99, arc_length_parametrization));
+    }
```

Comment:
Did we make any changes to this file explicitly? If not we should probably revert all changes to it, indenting seems odd here.

Reply:
Reverted changes